cmd/preview/cmdpreview.go

```diff
@@ -76,26 +88,27 @@ func NewCmdPreview(f cmdutil.Factory, ioStreams genericclioptions.IOStreams) *co
 				ch = applier.Run(ctx, infos, apply.Options{
 					EmitStatusEvents: false,
 					NoPrune:          noPrune,
-					DryRun:           true,
+					DryRunStrategy:   drs,
 				})
 			} else {
 				ch = destroyer.Run()
 			}
 
 			// The printer will print updates from the channel. It will block
 			// until the channel is closed.
-			printer.Print(ch, true)
+			printer.Print(ch, drs)
 		},
 	}
 
 	cmd.Flags().BoolVar(&noPrune, "no-prune", noPrune, "If true, do not prune previously applied objects.")
+	cmd.Flags().BoolVar(&serverDryRun, "server", serverDryRun, "If true, preview runs in the server instead of the client.")
```

Comment:
We can't use `server` as there is already a global flag with this name. Maybe we can use `server-side` as it would follow the convention for `kubectl apply`.

Reply:
Done. Let me know how this looks. This would set the server-side flag directly on kubectl library.